### ISSUE TITLE
Add -W flag to lvcreate from TUI

### ISSUE THREAD

**Issue by mastergregor** [parodyne/esos#225]

**Comment by msmith626** [comment]
-W|--wipesignatures {y|n}
              Controls detection and subsequent wiping of signatures on  newly
              created  Logical  Volume.  There's  a  prompt for each signature
              detected to confirm its wiping (unless --yes is used  where  LVM
              assumes  'yes'  answer  for  each prompt automatically). If this
              option is not specified, then by default -W | --wipesignatures y
              is  assumed  each time the zeroing is done (-Z | --zero y). This
              default      behaviour      can      be      controlled       by
              allocation/wipe_signatures_when_zeroing_new_lvs setting found in
              lvm.conf(5).
              If blkid wiping is used (allocation/use_blkid_wiping setting  in
              lvm.conf(5))  and  LVM2  is  compiled with blkid wiping support,
              then blkid(8) library is used  to  detect  the  signatures  (use
              blkid  -k  command  to list the signatures that are recognized).
              Otherwise, native LVM2 code is used  to  detect  signatures  (MD
              RAID, swap and LUKS signatures are detected only in this case).
              Logical volume is not wiped if the read only flag is set.

Can you give some background / usage that adding the '-W' flag would
help you overcome or handle? This is the first request we've had for
this...

**Comment by msmith626** [comment]
I agree the possible interactive behavior when running the commands needs
to be solved, but I'm not sure I agree with using '--yes' to always wipe
the device. I expect this could be safety catch, so probably don't want to
wipe unconditionally.

What about if we just get it to return an error if this can't be done...
then the user can try running at at the shell and determine how they'd like
to handle it.

-q|--quiet
              Suppress output and log messages.  Overrides -d and -v.
Repeat once to also suppress any prompts with answer 'no'.

This option might be what we want in the TUI... answer "no" to any
interactive prompts automatically. It would be ideal if this also returns a
non-zero status. Any chance you could test this behavior on another
non-production system?

--Marc

**Comment by mastergregor** [comment]
Quiet might be a good option - I wanted to suggest that lvcreate can fail, and then TUI can prompt for confirmation of a whipeout and do it again with -yes, but after some coffee, I think that would only open the door for potential strange behaviors from the unknown situations as this point. So I do like a suggestion to just error out, and let user do it manually, since sensitive systems like storage should be handled by people who know what they are doing anyhow.
As for testing, it will be a month or two before I can test it properly, I am thinking of getting a sister enclosure for HA setup, but this has to wait for Q1-2019 funding, since I would need an FC switch at that time as well.
It is a Christmas season anyhow, time for skiing and family :)

Gregor

    On Monday, December 10, 2018, 9:48:54 AM CST, Marc A. Smith <notifications@github.com> wrote:  
 
 I agree the possible interactive behavior when running the commands needs
to be solved, but I'm not sure I agree with using '--yes' to always wipe
the device. I expect this could be safety catch, so probably don't want to
wipe unconditionally.

What about if we just get it to return an error if this can't be done...
then the user can try running at at the shell and determine how they'd like
to handle it.

-q|--quiet
 Suppress output and log messages. Overrides -d and -v.
Repeat once to also suppress any prompts with answer 'no'.

This option might be what we want in the TUI... answer "no" to any
interactive prompts automatically. It would be ideal if this also returns a
non-zero status. Any chance you could test this behavior on another
non-production system?

--Marc

**Comment by msmith626** [comment]
Okay thanks, I'll leave this open for now. I hope to have some time in the
coming weeks to be able to reproduce this and test the '--quiet' flag. If
it returns an exit status of 0 when input is needed, then this is no good
our situation. So we'll check that first. =)

--Marc

**Comment by mastergregor** [comment]
I got around trying a few things, but I think results are not really what is wanted here:

![lvcreatecapture](https://user-images.githubusercontent.com/43911786/50405830-0d307a80-0780-11e9-9923-f48049866663.PNG)

It looks like "-q -q" makes lvcreate perform as expected, and it completes "successfully" :)

I will keep this setup at it's present state for now, so we can try out different options for lvcreate when there is an existing signature on the volume space.

Ideas?

**Comment by mastergregor** [comment]
I did not really make it clear in previous post that "lvcreate -q -q" does create a logical volume, but it does not wipe the existing signature of a previous file system on it. 

![capture1](https://user-images.githubusercontent.com/43911786/50451581-91a70880-08fa-11e9-9910-fda996b3873b.PNG)

I feel this was an important part of information to supply :)

In my case I did not mind that there was an existing FS on it, since I was putting an ext4 partition on it anyhow, but others may want a different behavior in certain situations.

**Comment by msmith626** [comment]
Thank you for your work in testing this. The "-q -q" makes lvcreate (or any
LVM command) simply answer "no" when prompted. This may not be the best in
every situation either, especially if it answers "no" to creating a logical
volume (for something other than wiping data on the volume), and then
volume doesn't get created, and it still exits with 0.

That said, I think forcing an answer of "no" for all LVM commands is
probably better than us hanging at the shell waiting for input. But it
would be nice to know other cases where the logical volume didn't actually
get created if a "no" answer was forced, and we would want the command to
exit non-zero.

Let me think about this a little more...
